pkg/agent/multicast/mcast_controller.go

```diff
@@ -181,13 +181,15 @@ func (c *Controller) clearStaleGroups() {
 	for _, obj := range c.groupCache.List() {
 		status := obj.(*GroupMemberStatus)
 		diff := now.Sub(status.lastIGMPReport)
-		if diff > c.mcastGroupTimeout {
-			// Notify worker to remove the group from groupCache if all its members are not updated before mcastGroupTimeout.
-			c.queue.Add(status.group.String())
+		if diff > c.mcastGroupTimeout && len(status.localMembers) == 0 {
+			c.checkLastMember(status.group)
+			klog.InfoS("Group is timeout without any members", "group", status.group.String(), "timeDiff", diff.String())
```

Comment:
I don't think explicitly calling the `String` method is necessary here

Reply:
removed.